docs/src/pages/[...docPath].tsx

```diff
@@ -0,0 +1,130 @@
+import type { NextPage, GetStaticProps, GetStaticPaths } from "next";
+import fs from "fs";
+import matter from "gray-matter";
+import { MDXRemote } from "next-mdx-remote";
+import { serialize } from "next-mdx-remote/serialize";
+import remarkDirective from "remark-directive";
+import { visit } from "unist-util-visit";
+import { h } from "hastscript";
+
+import {
+  getMDPaths,
+  withIndexFile,
+  withIndexURL,
+  withInsertedCodeFromLinks,
+  withoutComments,
+} from "../model/md-generate";
+import Title from "../components/mdxComponents/Title";
+import Paragraph from "../components/mdxComponents/Paragraph";
+import CodeBlocks from "../components/mdxComponents/CodeBlocks";
+import Admonition from "../components/mdxComponents/Admonition";
+import DocumentationLayout from "../components/DocumentationLayout";
+
+const components = {
+  h2: Title.H2,
+  h3: Title.H3,
+  p: Paragraph,
+  code: CodeBlocks.Code,
+  pre: CodeBlocks.Pre,
+  tip: Admonition.Tip,
+  warning: Admonition.Warning,
+};
+
+/** @type {import('unified').Plugin<[], import('mdast').Root>} */
+function myRemarkPlugin() {
+  // @ts-ignore
+  return (tree) => {
+    visit(tree, (node) => {
+      if (
+        node.type === "textDirective" ||
+        node.type === "leafDirective" ||
+        node.type === "containerDirective"
+      ) {
+        const data = node.data || {};
+        const hast = h(node.name, node.attributes);
+        // @ts-ignore
+        data.hName = hast.tagName;
```

Comment:
Looks like it's the case where it's worth adding some comments about where "tip" and "warnings" are translated into node

Reply:
done